### PR TITLE
[Feature] : Get in touch component

### PR DESCRIPTION
**Pull request by SahilVermaFED**
## Description

-   [Ticket Link]
-  Get in touch component afterwards will attach to footer section while making homepage with background design with css 

- ## UI screenshot or Video
[Screencast from 04-03-24 05:41:42 PM IST.webm](https://github.com/primathonacademy/april-batch-live-project/assets/131689244/45da3f2c-b186-49b4-8934-66c4c81de07a)

### General:

-   [x] **PR Size :** Confirming PR size < 20 files or pre-approved if > 20 files.
-   [x] **Self Review :** Confirming I have self-reviewed my PR
-   [x] **Peers Review :** - [ ] @Dev 1
-   [x] Confirming I have added all sensitive information-related should be inside .env files

### Client:

-   [x] **Console :** Confirming no console errors in the UI for this update.

-   [x] **Import order :** Confirming correct import order.

```
# External Import
import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';

# Internal Import
import { } from '../../Api/ApiRoutes';
import { } from '../../constants/AppConstant';
```

-   [x] **Re-renders :** Confirming minimal re-renders for new React components.

-   [x] **Network calls :** Confirming efficient network/API calls, minimized as needed.

-   [x] **Util function comments :** Confirming comprehensive comments for utility functions.

-   [x] **Image optimization :** Confirming I have checked that all images are optimized. If not, please use https://tinyjpg.com/ or https://vecta.io/nano.

-   [x] **Dead code :** Confirming I have removed all comments code which is of no use OR written why I have kept the commented code.

### Confirming different viewport testing of the entire flow:

-   [x] Tested on 280px viewport
-   [x] Tested on 320px viewport
-   [x] Tested on 375px viewport
-   [x] Tested on 425px viewport
-   [x] Tested on 768px viewport
-   [x] Tested on 1024px viewport
-   [x] Tested on 1440px viewport
-   [x] Tested on 2560px viewport

### End-to-end testing:

-   [x] **Happy flow :** Successfully tested end-to-end happy flow, all working as expected.
-   [x] **Error flow :** Tested UI flow with error scenarios, gracefully handled errors on UI
-   [x] **Common component changed :** Tested change in common component across views to prevent breaking elsewhere.

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -12,19 +12,59 @@ export default {
...
-                '5xl': '2rem'
+                '5xl': '2rem',
+                '6xl': '2.25rem',
+                '11xl': '3.50rem'
+            },
+            width: {
+                180: '45rem',
+                95: '23.75rem',
+                100: '25rem',
+                137.5: '550px'
```
Comment (ram-primathon) [inline tailwind.config.js:38]: why px?

File: tailwind.config.js [hunk 2]
```diff
@@ -12,19 +12,59 @@ export default {
...
+            },
+            lineHeight: {
+                tighter: '4rem',
+                12.25: '49px',
```
Comment (ram-primathon) [inline tailwind.config.js:42]: why px?

File: tailwind.config.js [hunk 3]
```diff
@@ -12,19 +12,59 @@ export default {
...
+                16: '64px',
```
Comment (ram-primathon) [inline tailwind.config.js:43]: why px?

File: tailwind.config.js [hunk 4]
```diff
@@ -12,19 +12,59 @@ export default {
...
+                18: '72px'
```
Comment (ram-primathon) [inline tailwind.config.js:44]: why px?

File: tailwind.config.js [hunk 5]
```diff
@@ -12,19 +12,59 @@ export default {
...
+            },
+            padding: {
+                45: '11.25rem',
+                90: '22.5rem'
             },
             fontSize: {
-                xxs: '0.625rem'
+                xxs: '0.625rem',
+                '7.5xl': '80px'
```
Comment (ram-primathon) [inline tailwind.config.js:52]: why px?

File: tailwind.config.js [hunk 6]
```diff
@@ -12,19 +12,59 @@ export default {
...
+            },
+            screens: {
+                '2xs': '375px',
```
Comment (ram-primathon) [inline tailwind.config.js:55]: why px?

File: tailwind.config.js [hunk 7]
```diff
@@ -12,19 +12,59 @@ export default {
...
+                xs: '425px',
```
Comment (ram-primathon) [inline tailwind.config.js:56]: why px?

File: tailwind.config.js [hunk 8]
```diff
@@ -12,19 +12,59 @@ export default {
...
+                '3xl': '1768px'
```
Comment (ram-primathon) [inline tailwind.config.js:57]: why px?

File: tailwind.config.js (hunk 6 above)
Comment (ram-primathon) [reply tailwind.config.js:55]: Why we need this?

File: tailwind.config.js (hunk 7 above)
Comment (ram-primathon) [reply tailwind.config.js:56]: Why we need this?

File: tailwind.config.js (hunk 8 above)
Comment (ram-primathon) [reply tailwind.config.js:57]: Why we need this?

File: tailwind.config.js [hunk 9]
```diff
@@ -12,19 +12,59 @@ export default {
...
+            },
+            spacing: {
```
Comment (ram-primathon) [inline tailwind.config.js:59]: delete this you can't use this

File: tailwind.config.js [hunk 10]
```diff
@@ -12,19 +12,59 @@ export default {
...
+                'fade-blue-white': '#F4F8FF'
             },
             borderRadius: {
                 '4xl': '1.75rem',
-                '5xl': '2rem'
+                '5xl': '2rem',
+                '6xl': '2.25rem',
+                '11xl': '3.50rem'
+            },
+            width: {
```
Comment (ram-primathon) [inline tailwind.config.js:34]: Why are you adding so many configuration changes?

**Review by ram-primathon — CHANGES_REQUESTED** [review]
This is a very bad PR

File: src/pages/homepage/GetInTouch.jsx [hunk 11]
```diff
@@ -2,26 +2,39 @@ import { NavLink } from 'react-router-dom';
...
+
     return (
-        <div className="px-9 md:px-16 lg:px-20 xl:px-28 2xl:px-36 py-16 md:py-24 lg:py-32 bg-light-pink">
-            <section className="flex flex-col md:flex-row w-full bg-accent-blue p-3 rounded-lg relative">
-                <div className="flex flex-col gap-1 justify-center xl:items-start py-16 xl:py-20 2xl:py-24 3xl:py-32 xl:px-28 3xl:px-36 md:gap-0 xl:gap-2 2xl:gap-4 items-center w-full relative">
-                    <p className="font-poppins text-white opacity-80 text-start text-sm sm:text-base lg:text-2xl xl:text-3xl 2xl:text-4xl w-44 sm:w-48 lg:w-72 xl:w-full md:leading-6 lg:leading-8">
+        <div className="px-6 xs:px-9 md:px-16 lg:px-20 xl:px-28 2xl:px-36 py-16 md:py-24 lg:py-32 bg-fade-white">
+            <section className="flex flex-col md:flex-row w-full bg-accent-blue p-3 rounded-lg md:rounded-xl relative overflow-hidden">
+                <div className="absolute left-0 top-28 sm:top-31 lg:top-20 xl:top-39.5 2xl:top-34.5 3xl:top-43.5 flex flex-col gap-11 lg:gap-12.5 xl:gap-16.5 2xl:gap-19.5 3xl:gap-24 w-full border-fade-white opacity-20">
+                    <hr />
```
Comment (ram-primathon) [inline src/pages/homepage/GetInTouch.jsx:13]: Why do you need 4 hr?

File: tailwind.config.js (hunk 10 above)
Comment (TheSahilVerma) [reply tailwind.config.js:34]: i have added just only for bg im removing all these

File: tailwind.config.js (hunk 8 above)
Comment (TheSahilVerma) [reply tailwind.config.js:57]: now we are not using this we are using smaller screen size for actual design with which code doesn't break

**Pull request closed without merging**